docs/architecture/adr-083-libp2p-implementation.md

```diff
@@ -0,0 +1,163 @@
+# ADR 80: Libp2p Implementation Plan
+
+## Changelog
+
+- 2022-05-20: Initial Draft.
+
+## Status
+
+Accepted
+
+## Context
+
+Adopting libp2p is a larger project that is the culmination of a long
+programme of work in the Tendermint codebase, and this document
+describes some of the decisions related to that adoption.
+
+## Alternative Approaches
+
+None. 
+
+The decision to adopt libp2p was part of ADR 73, and this ADR lays out
+an implementation plan and records a collection of decisions related
+to this implementation which have not been formally collected in any
+other form. 
+
+## Decisions
+
+### Feature Parity
+
+The goal of this project should be a p2p stack that has feature parity
+with the legacy stack. Everything that is possible with the p2p stack
+today should be possible in the future, eventually reducing the
+overhead of peer communication and complexity in the gossip/peer layer
+of the system. 
+
+In persuit of this goal, this project has two explicit *non-goals*:
+
+- Do not require wire compatibility with legacy stack.
+- Do not support dual-mode (legacy/libp2p) in a single node or network.
+
+There does not seem to be a compelling use case of having a release
+that contains both P2P stacks. While either of these goals could
+be theoretically possible, by eliminating both of these goals we can
+effectively reduce the scope of the project and increase our
+confidence.
+
+### Target Iterative Progress 
+
+Rather than produce an extensive design document (other than this
+record of decisions,) we will work iteratively in a topic
+branch--[`main/libp2p`](https://github.com/tendermint/tendermint/tree/main/libp2p)--to
+prototype the new libp2p integration. This work will aim to:
```

Comment:
This is a substantial concern for me in that it implies that we don't know yet where we're going with the libp2p refactor, and as such from a project planning perspective it's impossible to scope it (and therefore it's far harder to divide up the work, since we don't know what we're dividing up) or de-risk it.